File: 01/a.py
with open("./input.txt") as f:
    arr = [int(i) for i in f.readlines()]
    for i in range(len(arr)-1):
        for j in range(i+1, len(arr)):
            if arr[i] + arr[j] == 2020:
                print(arr[i] * arr[j])
                break

File: 01/A.hs
subsets 0 _ = [[]]
subsets _ [] = []
subsets n (x : xs) = map (x :) (subsets (n - 1) xs) ++ subsets n xs

main = do
    contents <- fmap lines $ readFile "input.txt"
    let d = map read contents :: [Int]
    print $ [product x | x <- subsets 2 d, sum x == 2020] !! 0